code/ship/ship.h

```diff
@@ -714,7 +714,9 @@ class ship
 	void clear();
 
     //Helper functions
-    inline bool is_arriving() { return flags[Ship::Ship_Flags::Arriving_stage_1, Ship::Ship_Flags::Arriving_stage_2]; }
+    inline bool is_arriving() { return flags[Ship::Ship_Flags::Arriving_stage_1, Ship::Ship_Flags::Arriving_stage_1_ndl, Ship::Ship_Flags::Arriving_stage_2, Ship::Ship_Flags::Arriving_stage_2_ndl]; }
+    inline bool is_arriving_dock_leader() { return flags[Ship::Ship_Flags::Arriving_stage_1, Ship::Ship_Flags::Arriving_stage_2]; }
+    inline bool is_arriving_stage_1_all_docked() { return flags[Ship::Ship_Flags::Arriving_stage_1, Ship::Ship_Flags::Arriving_stage_1_ndl]; }
```

Comment:
I'm happy to rename them, however I just want to confirm that I understand!

`is_arriving_actively()` means two flags; both stage 1 & 2 but not the followers
`is_arriving_dock_follower()` means stage one only; but everyone
`is_arriving()` means four flags; everyone in both stages

thanks for the review!

Reply:
Uh... rats, I didn't even consider the distinction between stage 1 and 2.

I think it really is necessary to make this a function with parameters rather than an inline function.  One parameter should be stage1/stage2/both, and one parameter should be dock-leader-or-solo/dock-follower/dock-agnostic.

Reply:
done, could you please let me know what you think?